explainer.md

```diff
@@ -516,22 +527,147 @@ function onSessionStarted(session) {
+  // Use the device pose to create a ray from the head, regardless of whether controllers are connected.
+  let devicePose = event.frame.getDevicePose(xrFrameOfRef);
+
+  // Ray cast into scene with the device pose to determine if anything was hit.
+  // Assumes the use of a fictionalized math and scene library.
+  let rayOrigin = getTranslation(devicePose.poseModelMatrix);
+  let rayDirection = applyRotation(scene.forwardVector, devicePose.poseModelMatrix);
+  let selectedObject = scene.getObjectIntersectingRay(rayOrigin, rayDirection);
+  if (selectedObject) {
+    selectedObject.onSelect();
+  }
+}
+```
+
+### Rendering input sources
+
+Most applications will want to visually represent the input sources somehow. The appropriate type of visualization to be used depends on the value of the `targetRayMode` attribute:
+
+  * `'gazing'`: A cursor should be drawn at some distance down the target ray, ideally at the depth of the first surface it intersects with, so the user can identify what will be interacted with when a select event is fired. It's not appropriate to draw a controller or ray in this case, since they may obscure the user's vision or be difficult to visually converge on.
+  * `'pointing'`: If the `gripMatrix` in not `null` an application-appropriate controller model should be drawn using that matrix as the transform. If appropriate for the experience, the a visualization of the target ray and a cursor as described in the `'gazing'` should also be drawn.
+  * `'tapping'`: In all cases the point of origin of the target ray is obvious and no visualization is needed.
+
+```js
+// These methods presumes the use of a fictionalized rendering library.
+
+// Render a visualization of the input source - eg. a controller mesh.
+function renderInputSource(session, inputSource, inputPose) {
+  // FIXME: Using a fictional isDisplayOpaque() method to state that controller meshes should not be rendered
+  // on transparent displays (AR).
+  if (isDisplayOpaque(session) && inputPose.gripMatrix) {
+    // Render a controller mesh if the using the gripMatrix as a transform.
+    let controllerMesh = getControllerMesh(inputSource);
+    renderer.drawMeshWithTransform(controllerMesh, inputPose.gripMatrix);
+  }
+}
+
+// Render a visualization of target ray of the input source - eg. a line or cursor.
+// Presumes the use of a fictionalized rendering library.
+function renderCursor(inputSource, inputPose) {
+  // Only render a target ray if this was the most recently used input source.
+  if (inputSource.targetRayMode == "pointing") {
+    // Draw targeting rays for pointing devices only.
+    renderer.drawRay(inputPose.targetRayMatrix);
+  }
+
+  if (inputSource.targetRayMode != "tapping") {
+    // Draw a cursor for gazing and pointing devices only.
+    let cursorPosition = scene.getIntersectionPoint(inputPose.targetRayMatrix);
+    if (cursorPosition) {
+      renderer.drawCursor(cursorPosition);
+    }
+  }
+}
+```
+
+### Complex input
```

Comment:
"Complex input" is maybe not the phrase I would use here. Maybe "Direct manipulation" or "Grabbing and dragging"?